docs/parser.md

```
@@ -0,0 +1,175 @@
+may be guarded.)
+
+## Parsing
+
+The most general way to match a parser against a stream is the `parse(parser, stream=)`
+function, which either
+
+* returns an array of outputs and advances `stream` (if the parser matches), or
+* returns `None` and leaves `stream` unchanged (if the parser does not match).
+
+For example:
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)